### PR TITLE
Fixes #173: Implement Country code use case

### LINKED ISSUE #173: Implement country code use cases
**Is your feature request related to a problem? Please describe.**
Country codes should be extracted from GTFS datasets using `stop_lat` and `stop_lon` from `stops.txt`

**Describe the solution you'd like**
Implement [reverser_geocoder](https://pypi.org/project/reverse_geocoder/) to get country codes

### PR DESCRIPTION
**Pull request by maximearmstrong**
**Summary:**

Fixes #173: Implement Country code use case

This PR implement the Country code use case by leveraging the [reverse_geocoder](https://pypi.org/project/reverse_geocoder/) python library

Implement

- `process_country_codes_for_gtfs_metadata.py` and tests

This PR contains changes to

- `main.py`, `gtfs_metadata.py` and `create_dataset_entity_for_gtfs_metadata.py` were modified to reflect changes
- `.env.staging` and `.env.production` were modified to add the country code property

**Expected behavior:** 

Country codes are processed during the metadata extraction.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `pytest` to make sure you didn't break anything
- [x] Format the title like "<short description of fix and changes>" (for example - "Check for null value before using field")
- [x] Linked all relevant issues
- [ ] ~Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -46,7 +46,9 @@ python-dotenv==0.15.0
 pytz==2020.1
 rdflib==5.0.0
 requests==2.24.0
+reverse-geocoder==1.5.1
 Rtree==0.9.4
+scipy==1.6.2
```
Comment (annaelleduff) [inline requirements.txt:51]: why is scipy added? I might have missed an import but I don't see it

File: .env.production [hunk 2]
```diff
@@ -7,6 +7,7 @@ GTFS_SCHEDULE_DATA_FORMAT = ""
 INSTANCE_PROP = ""
 SOURCE_ENTITY_PROP = ""
 MAIN_TIMEZONE_PROP = ""
+COUNTRY_CODE_PROP = "P8"
```
Comment (lionel-nj) [inline .env.production:10]: I might be missing something here, but should this be `""`?

Comment (maximearmstrong) [reply .env.production:10]: This is normal, all the other ones are fixed in #170

File: usecase/test/test_process_country_codes_for_gtfs_metadata.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        type(mock_gtfs_representation).dataset = mock_dataset
+        type(mock_gtfs_representation).metadata = mock_metadata
+
+        under_test = process_country_codes_for_gtfs_metadata(mock_gtfs_representation)
+        self.assertIsInstance(under_test, GtfsRepresentation)
+        mock_rg.assert_not_called()
+        self.assertEqual(mock_metadata.country_codes, set())
+
+    @mock.patch("usecase.process_country_codes_for_gtfs_metadata.rg.search")
+    def test_process_country_codes_with_lat_lon(self, mock_rg):
```
Comment (lionel-nj) [inline usecase/test/test_process_country_codes_for_gtfs_metadata.py:53]: One more unit test could consist in using out of range values for latitudes and longitudes: (lat=850, lon=-4000). The resulting set of country codes should be empty.

**Review by lionel-nj — CHANGES_REQUESTED** [review]
Overall, LGTM! Just one additional unit test could be implemented to make sure this new use case works properly. 

Just making sure, if any row of the provided GTFS dataset has no lat or lon (i.e. the validation report contains a `MissingRequiredFieldNotice`  notice for latitude and/or longitude), this use case will not be called right?

File: usecase/test/test_process_country_codes_for_gtfs_metadata.py (hunk 3 above)
Comment (maximearmstrong) [reply usecase/test/test_process_country_codes_for_gtfs_metadata.py:53]: Ah! I thought `reverse_geocoder` would crash on this, but indeed it just returns nothing so I added the test. We want the use cases to fail on invalid value such as (lat=850, lon=-4000). This will be captured by validating the use cases with the Validator output in #172. For example, (lat=850, lon=-4000) will generated a `number_out_of_range` notice for file `stops.txt`, so the use case won't be kept after verification.

**Review by lionel-nj — APPROVED** [review]
LGTM 👍🏾
